underfs/swift/src/main/java/alluxio/underfs/swift/SwiftInputStream.java

```diff
@@ -83,6 +93,9 @@ public int read(byte[] b, int offset, int length) throws IOException {
     int read = mStream.read(b, offset, length);
     if (read != -1) {
       mPos += read;
+      if (mPos % getBlockSize() == 0) {
+        closeStream();
+      }
```

Comment:
scenario 1) wouldn't that result in a partial read of 10 bytes

scenario 2) my understanding is that the code maintains the following invariant: `mPos % blockSize == 0 => mStream == null`